packages/shared/src/net/retry.ts

```diff
@@ -25,10 +30,52 @@ const defaultOptions: ExecuteWithNetworkRetriesOptions = {
     "ECONNREFUSED", // (Connection refused): No connection could be made because the target machine actively refused it. This usually results from trying to connect to a service that is inactive on the foreign host.
     "ECONNRESET", //  (Connection reset by peer): A connection was forcibly closed by a peer. This normally results from a loss of the connection on the remote socket due to a timeout or reboot. Commonly encountered via the http and net modules.
   ],
-  httpStatusCodesToRetry: [429], // 429 Too Many Requests
+  httpStatusCodesToRetry: [tooManyRequestsStatus], // 429 Too Many Requests
   retryOnTimeout: false,
 };
 
+function getHttpStatusFromError(error: unknown): number | undefined {
+  return axios.isAxiosError(error) ? error.response?.status : undefined;
+}
+
+function getHttpCodeFromError(error: unknown): string | undefined {
+  return axios.isAxiosError(error) ? error.code : undefined;
+}
+
+/**
+ * Custom getTimeToWait function for network retries. Has special handling for 429 Too Many Requests. Else regular backoff.
+ * @param initialDelay The initial delay in milliseconds.
+ * @param backoffMultiplier The backoff multiplier.
+ * @param attempt The current attempt number.
+ * @param maxDelay The maximum delay in milliseconds.
+ * @param error The error that occurred.
+ * @returns The time to wait before the next retry.
+ */
+function networkGetTimeToWait({
+  initialDelay,
+  backoffMultiplier,
+  attempt,
+  maxDelay,
+  error,
```

Comment:
What are you doing with this error?

Reply:
we are choosing a different delay based on the error ...